src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -416,6 +419,26 @@ fn rustc(cx: &mut Context, unit: &Unit, exec: Arc<Executor>) -> CargoResult<Work
         }
         Ok(())
     }
+
+    // Add all custom environment variables present in `state` (after they've
+    // been put there by one of the `build_scripts`) to the command provided.
+    fn add_custom_env(rustc: &mut ProcessBuilder,
+                      build_state: &BuildMap,
+                      build_scripts: &BuildScripts,
+                      current_id: &PackageId) -> CargoResult<()> {
+        for key in build_scripts.to_link.iter() {
```

Comment:
`current_id` is just a `PackageId`, and to key into `BuildMap` I also a `cargo_rustc::Kind` which I'm not entirely clear about the purpose of. I picked `Kind::Host` following the other similar usages and it seems to work no worse than the previous version.